src/lib/storage/boot_requirements_strategies/prep.rb

```diff
@@ -34,27 +34,42 @@ module BootRequirementsStrategies
       class PReP < Base
         def needed_partitions
           volumes = super
-          volumes << prep_volume if prep_partition_missing?
+          volumes << prep_volume if prep_partition_required? && prep_partition_missing?
           volumes
         end
 
       protected
 
         def prep_partition_missing?
-          true	# #prep_partitions not implemented yet
-          # disk_analyzer.prep_partitions.empty?
+          partitions = disk_analyzer.prep_partitions[settings.root_device]
+          partitions.nil? || partitions.empty?
+        end
+
+        def prep_partition_required?
+          # no need of PReP partition in OPAL/PowerNV/Bare metal
+          !arch.power_nv?
         end
 
         def prep_volume
-          # TODO: we need to pass partition type somehow (PReP partition, ID_PPC_PREP, ID_GPT_PREP)
-          # TODO: hoe to really pass 'no mount point' and 'no fs'?
-          vol = PlannedVolume.new("", ::Storage::FsType_UNKNOWN)
-          vol.min_size = DiskSize.MiB(1)
+          vol = PlannedVolume.new(nil)
+          # So far we are always using msdos partition ids
+          vol.partition_id = ::Storage::ID_PPC_PREP
+          vol.min_size = DiskSize.kiB(256)
           vol.max_size = DiskSize.MiB(8)
           vol.desired_size = DiskSize.MiB(1)
+          # Make sure that alignment does not result in a too big partition
+          vol.align = :keep_size
+          vol.bootable = true
```

Comment:
I expect `PartitionCreator` to be smart enough to use `bootable` only in ms-dos partition tables. So far, all our code works only with ms-dos partition tables.